internal/pkg/install/mount.go

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package install
+
+import (
+	"log"
+
+	"github.com/autonomy/talos/internal/pkg/blockdevice"
+	"github.com/autonomy/talos/internal/pkg/blockdevice/probe"
+	gptpartition "github.com/autonomy/talos/internal/pkg/blockdevice/table/gpt/partition"
+	"github.com/autonomy/talos/internal/pkg/blockdevice/util"
+	"github.com/autonomy/talos/internal/pkg/constants"
+	"github.com/autonomy/talos/internal/pkg/mount"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+func Mount() (err error) {
+	log.Println("Discovering mountpoints")
+	var mp *mount.Points
+	if mp, err = mountpoints(); err != nil {
+		return errors.Errorf("error initializing block devices: %v", err)
+	}
+
+	/*
+		// This seems to be corrupting the filesystem
+		// Running xfs_repair finds a bunch of badness
+
+		log.Println("Repairing mountpoints")
+		for _, name := range []string{constants.RootPartitionLabel, constants.DataPartitionLabel, constants.BootPartitionLabel} {
+
+			if mountpoint, ok := mp.Get(name); ok {
+				if err = repair(mountpoint); err != nil {
+					return errors.Errorf("error fixing %s partition: %v", name, err)
+				}
+			}
+		}
+	*/
```

Comment:
FWIW this is only done to the `DATA` partition.
https://github.com/autonomy/talos/blob/ee226dddac3bd2a631093b44e05ce4725a39ed95/internal/app/init/internal/rootfs/mount/mount.go#L102-L106

Reply:
Good point. I was assuming that it _would_ be safe to run against each partition. Looks like the corruption happened when repairing the boot partiton.